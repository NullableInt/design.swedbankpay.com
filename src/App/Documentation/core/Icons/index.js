import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc } from "#";

const Usage = () => (
    <>
        <h2 id="usage">Usage</h2>
        <p>To use an icon, provide the following markup: <PrismCode className="language-html">{"<i class=\"material-icons\">[ICON NAME]</i>"}</PrismCode>.</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true}>
            <i className="material-icons">android</i>{"\n"}
            <i className="material-icons">contact_support</i>{"\n"}
            <i className="material-icons">face</i>{"\n"}
            <i className="material-icons">star</i>{"\n"}
            <i className="material-icons">warning</i>{"\n"}
        </ComponentPreview>
        {/* <h3>Sizes</h3>
        <p>Material icons also come in different sizes, the available ones are <code className="token property">.tiny</code>, <code className="token property">.small</code>, <code className="token property">.medium</code> and <code className="token property">.large</code>.</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true}>
            <i className="material-icons tiny">android</i>{"\n"}
            <i className="material-icons small">android</i>{"\n"}
            <i className="material-icons">android</i>{"\n"}
            <i className="material-icons medium">android</i>{"\n"}
            <i className="material-icons large">android</i>{"\n"}
        </ComponentPreview> */}
    </>
);

const IconsText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <p className="lead">The PayEx DesignGuide includes Google{"'"}s material icons. For a full overview of the available icons please visit <a href="https://material.io/tools/icons/">this site</a>.</p>
        <Usage />
    </div>
);

const Icons = () => {
    return (
        <div className="doc-container">
            <div className="row">
                <IconsText />
                <DocToc component={IconsText} />
            </div>
        </div>
    );
};

export default Icons;

/* for testing */
export { Usage, IconsText };
