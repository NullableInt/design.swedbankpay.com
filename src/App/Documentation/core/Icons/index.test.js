import React from "react";
import { shallow } from "enzyme";

import Icons, { Usage, IconsText } from "./index";

describe("Core: Icons", () => {
    it("is defined", () => {
        expect(Icons).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Icons />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("Usage", () => {
        it("is defined", () => {
            expect(Usage).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <Usage />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("IconsText", () => {
        it("is defined", () => {
            expect(IconsText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <IconsText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
