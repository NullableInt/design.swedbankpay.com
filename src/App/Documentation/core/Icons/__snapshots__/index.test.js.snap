// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: Icons IconsText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    The PayEx DesignGuide includes Google
    '
    s material icons. For a full overview of the available icons please visit 
    <a
      href="https://material.io/tools/icons/"
    >
      this site
    </a>
    .
  </p>
  <Usage />
</div>
`;

exports[`Core: Icons Usage renders 1`] = `
<React.Fragment>
  <h2
    id="usage"
  >
    Usage
  </h2>
  <p>
    To use an icon, provide the following markup: 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;i class="material-icons"&gt;[ICON NAME]&lt;/i&gt;
    </PrismCode>
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <i
      className="material-icons"
    >
      android
    </i>
    

    <i
      className="material-icons"
    >
      contact_support
    </i>
    

    <i
      className="material-icons"
    >
      face
    </i>
    

    <i
      className="material-icons"
    >
      star
    </i>
    

    <i
      className="material-icons"
    >
      warning
    </i>
    

  </ComponentPreview>
</React.Fragment>
`;

exports[`Core: Icons renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <IconsText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
